### ISSUE TITLE
x/tools/gopls: importing wrong package "golang.org/x/exp/slices"

### ISSUE THREAD

**Issue by jgardona** [golang/go#63641]
I don't know exactly if this is an issue of gopls, but I think its good to post it here.
When using slices package, the lsp will always import the wrong go package
**"golang.org/x/exp/slices"**, instead of the correct package **"slices"**.

### gopls version
```
Build info
----------
golang.org/x/tools/gopls v0.13.2
    golang.org/x/tools/gopls@v0.13.2 h1:Pyvx6MKvatbX3zzZmdGiFRfQZl0ohPlt2sFxO/5j6Ro=
    github.com/BurntSushi/toml@v1.2.1 h1:9F2/+DoOYIOksmaJFPw1tGFy1eDnIJXg+UHjuD8lTak=
    github.com/google/go-cmp@v0.5.9 h1:O2Tfq5qg4qc4AmwVlvv0oLiVAGB7enBSJ2x2DqQFi38=
    github.com/sergi/go-diff@v1.1.0 h1:we8PVUC3FE2uYfodKH/nBHMSetSfHDR6scGdBi+erh0=
    golang.org/x/exp@v0.0.0-20220722155223-a9213eeb770e h1:+WEEuIdZHnUeJJmEUjyYC2gfUMj69yZXw17EnHg/otA=
    golang.org/x/exp/typeparams@v0.0.0-20221212164502-fae10dda9338 h1:2O2DON6y3XMJiQRAS1UWU+54aec2uopH3x7MAiqGW6Y=
    golang.org/x/mod@v0.12.0 h1:rmsUpXtvNzj340zd98LZ4KntptpfRHwpFOHG188oHXc=
    golang.org/x/sync@v0.3.0 h1:ftCYgMx6zT/asHUrPw8BLLscYtGznsLAnjq5RH9P66E=
    golang.org/x/sys@v0.10.0 h1:SqMFp9UcQJZa+pmYuAKjd9xq1f0j5rLcDIk0mj4qAsA=
    golang.org/x/text@v0.11.0 h1:LAntKIrcmeSKERyiOh0XMV39LXS8IE9UL2yP7+f5ij4=
    golang.org/x/tools@v0.11.2-0.20230810185051-cc6b5804b8cf h1:Oush7UwPamr2/iNeNFBuNFj89YyHn0YY69EKDdvANnk=
    golang.org/x/vuln@v0.0.0-20230110180137-6ad3e3d07815 h1:A9kONVi4+AnuOr1dopsibH6hLi1Huy54cbeJxnq4vmU=
    honnef.co/go/tools@v0.4.2 h1:6qXr+R5w+ktL5UkwEbPp+fEvfyoMPche6GkOpGHZcLc=
    mvdan.cc/gofumpt@v0.4.0 h1:JVf4NN1mIpHogBj7ABpgOyZc65/UUOkKQFkoURsz4MM=
    mvdan.cc/xurls/v2@v2.4.0 h1:tzxjVAj+wSBmDcF6zBB7/myTy3gX9xvi8Tyr28AuQgc=
go: go1.21.3
```

### go env

GO111MODULE=''
GOARCH='amd64'
GOBIN=''
GOCACHE='/home/jgardona/snap/alacritty/common/.cache/go-build'
GOENV='/home/jgardona/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMODCACHE='/home/jgardona/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/home/jgardona/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/home/jgardona/Bin/go'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/home/jgardona/Bin/go/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.21.3'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='gcc'
CXX='g++'
CGO_ENABLED='1'
GOMOD='/dev/null'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build251818100=/tmp/go-build -gno-record-gcc-switches'

### What did you do?

Use slices package and wait for the gopls autocomplete works

### What did you expect to see?
Import "slices" package

### What did you see instead?
It imports "golang.org/x/exp/slices" package

### Editor and settings
* **vscode info**
Version: 1.83.1
Commit: f1b07bd25dfad64b0167beb15359ae573aecd2cc
Date: 2023-10-10T23:45:31.402Z
Electron: 25.8.4
ElectronBuildId: 24154031
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 6.5.0-9-generic snap

* no settings

![image](https://github.com/golang/go/assets/146892539/8428800d-197e-4760-bfac-f8615d2d195c)

**Comment by seankhliao** [comment]
what's in your go.mod?

**Comment by jgardona** [comment]
> what's in your go.mod?

example on  master [?] is 📦 v0.1.0 via 🦀 v1.73.0
❯ cat ../../go/example/go.mod
module example

go 1.21.3

**Comment by danp** [comment]
@jgardona did you end up with exp/slices as part of a completion or as part of "organize imports" (possibly on save)?

In other words, was it from doing something like:

```
func main() {
  slic<> // cursor at <> and golang.org/x/exp/slices suggested
}
```

or:

```
package main

func main() {
	slices.Sort([]int{5, 4, 3}) // hit save after typing this and `import "golang.org/x/exp/slices"` appears above
}
```

**Comment by jgardona** [comment]
@danp Hello. Good afternoon.

Exactly. I typed "slices.", then hit enter and  "golang.org/x/exp/slices" was imported.

**Comment by danp** [comment]
I see you're using `golang.org/x/tools/gopls v0.13.2`.

I believe the issue is that gopls embeds a list of stdlib packages and their exported symbols. That list is used by the completion bits to understand what stdlib packages are available. You can see the version of that list gopls v0.13.2 uses [here](https://github.com/golang/tools/blob/2f9d82ff532e4e725d7cf0f44aa0d5bacd283e18/internal/imports/zstdlib.go). There's no entry for the stdlib slices package since list was generated using info from Go 1.20 (as mentioned in the commit message).

To get a version of gopls that uses [the more recent list generated from Go 1.21](https://github.com/golang/tools/blob/732ad6f9f8bf74d273ff8a0dd7ab9b6c5bd425c6/internal/imports/zstdlib.go) (which does include [slices](https://github.com/golang/tools/blob/732ad6f9f8bf74d273ff8a0dd7ab9b6c5bd425c6/internal/imports/zstdlib.go#L5336)) you'd need to try a gopls v0.14.0 prerelease, such as [gopls/v0.14.0-pre.4](https://github.com/golang/tools/releases/tag/gopls%2Fv0.14.0-pre.4).

You should be able to try it with the example `go install` command there. If you're unable to try it, I think you'll need to wait for a proper v0.14.0 release.

And hopefully someone from the gopls will correct me if I've gotten something wrong!

**Comment by jgardona** [comment]
I'll wait gopls 0.14.0 and see what will be the output.

**Comment by hyangah** [comment]
gopls v0.14.0-pre.5 is available for testing. At least, it seems to work for me and suggests `slices` as the very first one. (the second one is the one in `golang.org/x/exp/slices`, but my project already depends on `golang.org/x/exp`.

![Screenshot 2023-10-24 at 5 00 18 PM](https://github.com/golang/go/assets/4999471/f9a83ad0-c623-4c27-aa12-39c2582c5dd7)

**Comment by findleyr** [comment]
gopls v0.14.0 is out now. @jgardona please let us know if this resolves your problem.

**Comment by jgardona** [comment]
> gopls v0.14.0 is out now. @jgardona please let us know if this resolves your problem.

confirmed here it was fixed. Thank you all :slightly_smiling_face:

**Comment by jgardona** [comment]
will close as it was fixed in 0.14
